### PR TITLE
"unbiased" std and Bessel's Correction for BatchNorm2d

### PR DESCRIPTION
**Pull request by symlons**
Reference to #774

### CODE REVIEW

File: tinygrad/nn/__init__.py [hunk 1]
```diff
@@ -18,7 +18,8 @@ def __call__(self, x:Tensor):
       # There's "online" algorithms that fix this, like https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_Online_algorithm
       batch_mean = x.mean(axis=(0,2,3))
       y = (x - batch_mean.reshape(shape=[1, -1, 1, 1]))
-      batch_var = (y*y).mean(axis=(0,2,3))
+      batch_var = y.std(axis=(0,2,3), unbiased=self.unbiased)
```
Comment (geohot) [inline tinygrad/nn/__init__.py:21]: Doing extra work.

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -309,10 +309,9 @@ def min(self, axis=None, keepdim=False): return -((-self).max(axis=axis, keepdim
   def mean(self, axis=None, keepdim=False):
     out = self.sum(axis=axis, keepdim=keepdim)
     return out * (prod(out.shape)/prod(self.shape))
-  # TODO: implement unbiased True option for torch bessel's correction (subtracting 1 from divisor causes 0.01 error)
-  def std(self, axis=None, keepdim=False):
+  def std(self, axis=None, keepdim=False, unbiased=False):
```
Comment (geohot) [inline tinygrad/tensor.py:312]: Please add tests

Comment (symlons) [reply tinygrad/tensor.py:312]: You mean somtething like that for std: 
helper_test_op([(45, 65, 85)], lambda x: torch.std(x, unbiased=True), lambda x: Tensor.std(x, unbiased=True))  ?

File: tinygrad/nn/__init__.py [hunk 3]
```diff
@@ -2,8 +2,8 @@
 from tinygrad.tensor import Tensor
 
 class BatchNorm2d:
-  def __init__(self, sz, eps=1e-5, affine=True, track_running_stats=True, momentum=0.1):
-    self.eps, self.track_running_stats, self.momentum = eps, track_running_stats, momentum
+  def __init__(self, sz, eps=1e-5, affine=True, track_running_stats=True, momentum=0.1, unbiased=False):
```
Comment (geohot) [inline tinygrad/nn/__init__.py:5]: Hmm, actually why is this being changed? It already matches torch.

Comment (symlons) [reply tinygrad/nn/__init__.py:5]: Pytorch used by default the biased estimator during training but the unbiased one for inference. According to this thread: https://github.com/pytorch/pytorch/issues/77427

File: tinygrad/tensor.py (hunk 2 above)
Comment (symlons) [reply tinygrad/tensor.py:312]: The test should already be included in test_std here: 7b689987182fedd482cc8879fe3935a12d813530

**Comment by geohot** [comment]
I will merge just the unbiased support for std, it's not a flag for batchnorm in torch. If you want, make a PR with just this and add more unbiased tests to test_ops.
